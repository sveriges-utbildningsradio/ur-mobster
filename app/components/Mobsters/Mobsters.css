.wrap {
  display: flex;
  flex-direction: column;
  margin: 25px 42px 30px;
  justify-content: space-between;
  height: 80vh;
}

.headerWrap {
  align-items: center;
  display: grid;
  grid-template-columns: 1fr 75px 20px;
}

h4 {
  color: #fff;
  font-family: 'Roboto', sans-serif;
  font-size: 26px;
  font-weight: 500;
}

.editLabel {
  margin-right: 10px;
  text-align: right;
}

.editButton {
  height: 24px;
  width: 24px;
}

.droppableWrap {
  min-height: 100px;
}

.mobster {
  display: grid;
  grid-template-columns: 45px 1fr 24px;
  grid-column-gap: 10px;
  height: 45px;
  margin-bottom: 15px;
}

.addButton {
  align-items: center;
  background: none;
  border: none;
  cursor: pointer;
  display: flex;
  height: 28px;
  justify-content: center;
  margin-left: 12px;
  width: 28px;
}

.addButton:focus {
  outline: 2px solid #2cdbec;
}

.addButtonImage {
  height: 24px;
  width: 24px;
}

.addWrap {
  display: grid;
  grid-template-columns: 45px 1fr;
  grid-column-gap: 10px;
}

.avatarWrap {
  display: inline-block;
  position: relative;
}

.avatar {
  border-radius: 50%;
  width: 45px;
}

.driverDot {
  height: 13px;
  width: 13px;
  background-color: #f70c54;
  border-radius: 50%;
  position: absolute;
  top: 0;
  left: 0;
}

.navigatorDot {
  height: 13px;
  width: 13px;
  background-color: #fff;
  border-radius: 50%;
  position: absolute;
  top: 0;
  left: 0;
}

.inputWrap {
  display: flex;
}

.name {
  color: #2cdbec;
  font-size: 18px;
  height: 28px;
  letter-spacing: normal;
  line-height: 1.56;
}

.githubName {
  font-size: 14px;
  font-weight: 500;
}

.userInput {
  background: rgb(17, 6, 54);
  border: none;
  border-bottom: 1px solid #2cdbec;
  color: #caf5ff;
  font-family: Roboto, sans-serif;
  font-size: 18px;
  margin-bottom: 5px;
  width: 100%;
}

input::-webkit-input-placeholder {
  color: #caf5ff;
  font-family: Roboto, sans-serif;
  font-size: 18px;
}

input:hover {
  background-color: rgb(17, 6, 54);
}

input:active,
input:focus {
  outline: none;
}
