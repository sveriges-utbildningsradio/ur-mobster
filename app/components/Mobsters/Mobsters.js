import React from 'react'
import styles from './Mobsters.css'
import avatarImage from '../../assets/avatar.png'
import githubButton from '../../assets/github.png'
import addButton from '../../assets/add.png'
import editButton from '../../assets/pencil.png'
import stopEditButton from '../../assets/pencil-red.png'
import removeButton from '../../assets/stop.png'

const Mobsters = ({
  activeUsers,
  clickEditButton,
  clickGitHubButton,
  clickGuestButton,
  clickRemoveUser,
  isEditing,
  setIsEditing,
  setUsername,
  username
}) => (
  <div className={styles.wrap}>
    <div>
      <div>
        <div className={styles.headerWrap}>
          <h4>Mobsters</h4>
          <img
            className={styles.editButton}
            onClick={clickEditButton}
            src={isEditing ? stopEditButton : editButton}
          />
        </div>
      </div>
      <ul>
        {!!activeUsers.length &&
          activeUsers.map(user => (
            <li className={styles.mobster} key={user.name}>
              <img
                className={styles.avatar}
                src={user.avatar ? user.avatar : avatarImage}
              />
              <div>
                <p className={styles.name}>{user.name}</p>
                <p className={styles.githubName}>{user.githubName}</p>
              </div>

              {isEditing && (
                <img
                  className={styles.editButton}
                  onClick={() => clickRemoveUser(user.name)}
                  src={removeButton}
                />
              )}
            </li>
          ))}
      </ul>
    </div>

    <div className={styles.addWrap}>
      <img className={styles.avatar} src={avatarImage} />
      <div>
        <div className={styles.inputWrap}>
          <input
            value={username}
            onChange={e => setUsername(e.target.value)}
            placeholder="Lägg till mobster"
          />
          <img
            className={styles.addButton}
            onClick={clickGuestButton}
            src={addButton}
          />
          <img
            className={styles.addButton}
            onClick={clickGitHubButton}
            src={githubButton}
          />
        </div>
        <p className={styles.githubName}>
          Skriv namn eller GitHub-användarnamn
        </p>
      </div>
    </div>
  </div>
)

export default Mobsters
