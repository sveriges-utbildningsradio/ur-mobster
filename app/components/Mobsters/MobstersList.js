// @flow

import React from 'react'
import { Droppable, Draggable } from 'react-beautiful-dnd'
import { FormattedMessage } from 'react-intl'
import type { User } from '../../types'
import styles from './Mobsters.css'
import avatarImage from '../../assets/avatar.png'
import removeButton from '../../assets/stop.png'

type MobstersListProps = {
  clickRemoveUser: (name: string, id: string) => void,
  droppableId: string,
  isEditing: boolean,
  users: User[]
}

const MobstersList = ({
  clickRemoveUser,
  droppableId,
  isEditing,
  users
}: MobstersListProps) => (
  <ul>
    <Droppable droppableId={droppableId}>
      {(provided, snapshot) => (
        /* eslint-disable react/jsx-props-no-spreading */
        <div
          {...provided.droppableProps}
          className={styles.droppableWrap}
          style={{
            backgroundColor: snapshot.isDraggingOver
              ? 'rgba(255, 255, 255, 0.1)'
              : 'rgb(17, 6, 54)'
          }}
          ref={provided.innerRef}
        >
          <ul data-e2e="mobsters-list-draggable-wrapper">
            {users && !!users.length ? (
              users.map((user, index) => (
                <Draggable
                  draggableId={user.name}
                  index={index}
                  key={user.id ? user.id : user.name}
                >
                  {providedDraggable => (
                    <li
                      className={styles.mobster}
                      key={user.id ? user.id : user.name}
                      ref={providedDraggable.innerRef}
                      {...providedDraggable.draggableProps}
                      {...providedDraggable.dragHandleProps}
                    >
                      <div className={styles.avatarWrap}>
                        <img
                          alt={user.name}
                          className={styles.avatar}
                          src={user.avatar ? user.avatar : avatarImage}
                        />
                        {index === 0 && droppableId === 'activeUsers' && (
                          <span className={styles.driverDot} />
                        )}
                        {index === 1 && droppableId === 'activeUsers' && (
                          <span className={styles.navigatorDot} />
                        )}
                      </div>
                      <div>
                        <p
                          className={styles.name}
                          data-e2e={`mobsterslist-name-${index}`}
                          data-testid={`mobsterslist-name-${index}`}
                        >
                          {user.name}
                        </p>
                        <p
                          className={styles.githubName}
                          data-e2e={`mobsterslist-githubName-${index}`}
                          data-testid={`mobsterslist-githubName-${index}`}
                        >
                          {user.githubName}
                        </p>
                      </div>

                      {isEditing && (
                        <input
                          alt={`Remove user from ${droppableId}`}
                          className={styles.editButton}
                          data-e2e="mobsterslist-remove-user"
                          data-testid={`mobsterslist-remove-${index}`}
                          onClick={() =>
                            clickRemoveUser(user.name, droppableId)
                          }
                          onKeyDown={e =>
                            e.keyCode === 13 &&
                            clickRemoveUser(user.name, droppableId)
                          }
                          src={removeButton}
                          type="image"
                        />
                      )}
                    </li>
                  )}
                </Draggable>
              ))
            ) : (
              <FormattedMessage id="noMobstersListItem" tagName="li" />
            )}
          </ul>
          {provided.placeholder}
        </div>
        /* eslint-enable react/jsx-props-no-spreading */
      )}
    </Droppable>
  </ul>
)

export default MobstersList
