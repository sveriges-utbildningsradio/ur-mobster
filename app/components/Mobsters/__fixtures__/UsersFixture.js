export const activeUsers = [
  {
    avatar: 'https://avatars1.githubusercontent.com/u/test',
    githubName: 'Active 1 Githubname',
    id: '2L_v21vEF',
    name: 'Active 1 Name'
  },
  { avatar: null, githubName: null, id: '9so97vODd', name: 'Active 2 Name' }
]

export const inactiveUsers = [
  {
    avatar: 'https://avatars1.githubusercontent.com/u/inactivetest',
    githubName: 'Inactive 1 Githubname',
    id: '2L_v21vEF',
    name: 'Inactive 2 Name'
  },
  { avatar: null, githubName: null, id: '9so97vODd', name: 'Inactive 2 Name' }
]
