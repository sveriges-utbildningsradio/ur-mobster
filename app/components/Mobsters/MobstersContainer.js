// @flow

import React, { useEffect, useReducer, useState } from 'react'
import axios from 'axios'
import storage from 'electron-json-storage'
import Mobsters from './Mobsters'
import generateMobsterName from '../../utils/generateMobsterName'
import { move, reorder } from '../../utils/listHelpers'

const setToStore = (list, users) => {
  storage.set(list, users, error => {
    if (error) throw error
  })
}

const reducer = (state, action) => {
  switch (action.type) {
    case 'ADD_ACTIVEUSER':
      setToStore('activeUsers', [...state.activeUsers, action.payload])
      return { ...state, activeUsers: [...state.activeUsers, action.payload] }
    case 'UPDATE_ACTIVEUSERS':
      setToStore('activeUsers', action.payload)
      return { ...state, activeUsers: action.payload }
    case 'UPDATE_INACTIVEUSERS':
      setToStore('inactiveUsers', action.payload)
      return { ...state, inactiveUsers: action.payload }
    case 'SET_FROM_STORAGE':
      return { ...action.payload }
    default:
      throw new Error()
  }
}

const initialState = {
  activeUsers: [],
  inactiveUsers: []
}

const MobstersContainer = ({ reachedEnd }: boolean) => {
  const init = ({ activeUsers, inactiveUsers }) => {
    storage.getAll((_, data) => {
      if (Object.keys(data).length === 0 && data.constructor === Object) {
        return { activeUsers, inactiveUsers }
      }

      dispatch({
        type: 'SET_FROM_STORAGE',
        payload: {
          activeUsers: data.activeUsers ? data.activeUsers : [],
          inactiveUsers: data.inactiveUsers ? data.inactiveUsers : []
        }
      })
    })

    return { activeUsers, inactiveUsers }
  }

  const [state, dispatch] = useReducer(reducer, initialState, init)
  const [username, setUsername] = useState('')
  const [isEditing, setIsEditing] = useState(false)

  useEffect(
    () => {
      if (reachedEnd === true) {
        const { activeUsers } = state
        const newMobsterOrder = activeUsers.concat(activeUsers.splice(0, 1))

        dispatch({ type: 'UPDATE_ACTIVEUSERS', payload: newMobsterOrder })
      }
    },
    [reachedEnd]
  )

  const getGitHubInfo = async () => {
    try {
      const { data } = await axios.get(
        `https://api.github.com/users/${username}`
      )

      const payload = {
        avatar: data.avatar_url,
        githubName: data.login,
        name: data.name ? data.name : generateMobsterName()
      }

      dispatch({
        type: 'ADD_ACTIVEUSER',
        payload
      })
    } catch (error) {
      console.error(error)
    }
  }

  const clickGitHubButton = () => {
    if (
      state.activeUsers.some(
        user =>
          user.githubName &&
          user.githubName.toLowerCase() === username.toLowerCase()
      )
    ) {
      return
    }
    getGitHubInfo()
    setUsername('')
  }

  const clickGuestButton = () => {
    if (
      state.activeUsers.some(
        user => user.name && user.name.toLowerCase() === username.toLowerCase()
      )
    ) {
      return
    }

    const payload = { avatar: null, githubName: null, name: username }

    dispatch({ type: 'ADD_ACTIVEUSER', payload })

    setUsername('')
  }

  const clickEditButton = () => {
    setIsEditing(!isEditing)
  }

  const clickRemoveUser = (userToRemove, list) => {
    const remainingUsers = state[list].filter(
      user => user.name !== userToRemove
    )

    if (list === 'activeUsers') {
      dispatch({
        type: 'UPDATE_ACTIVEUSERS',
        payload: remainingUsers
      })
    } else {
      dispatch({
        type: 'UPDATE_INACTIVEUSERS',
        payload: remainingUsers
      })
    }

    if (!remainingUsers.length) {
      setIsEditing(false)
      persistEmptyList(list)
    }
  }

  const persistEmptyList = list => {
    storage.set(list, [], error => {
      if (error) throw error
    })
  }

  const onDragEnd = result => {
    const { source, destination } = result

    // dropped outside the list
    if (!result.destination) {
      return
    }

    if (source.droppableId === destination.droppableId) {
      const items = reorder(
        state[source.droppableId],
        source.index,
        destination.index
      )

      if (source.droppableId === 'inactiveUsers') {
        dispatch({
          type: 'UPDATE_INACTIVEUSERS',
          payload: items
        })
      } else {
        dispatch({
          type: 'UPDATE_ACTIVEUSERS',
          payload: items
        })
      }
    } else {
      const sorted = move(
        state[source.droppableId],
        state[destination.droppableId],
        source,
        destination
      )

      dispatch({
        type: 'UPDATE_ACTIVEUSERS',
        payload: sorted.activeUsers
      })
      dispatch({
        type: 'UPDATE_INACTIVEUSERS',
        payload: sorted.inactiveUsers
      })
    }
  }

  return (
    <Mobsters
      activeUsers={state.activeUsers}
      clickEditButton={clickEditButton}
      clickGitHubButton={clickGitHubButton}
      clickGuestButton={clickGuestButton}
      clickRemoveUser={clickRemoveUser}
      inactiveUsers={state.inactiveUsers}
      isEditing={isEditing}
      onDragEnd={onDragEnd}
      setUsername={setUsername}
      username={username}
    />
  )
}

export default MobstersContainer
