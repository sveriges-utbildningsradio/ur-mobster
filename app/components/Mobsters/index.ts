import MobstersContainer from './MobstersContainer'
export default MobstersContainer
