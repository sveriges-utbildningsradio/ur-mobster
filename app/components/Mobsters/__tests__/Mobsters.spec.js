import React from 'react'
import { cleanup } from '@testing-library/react'
import 'jest-dom/extend-expect'
import renderWithReactIntl from '../../../utils/testHelpers'
import Mobsters from '../Mobsters'
import { activeUsers, inactiveUsers } from '../__fixtures__/UsersFixture'

describe('components/Mobsters', () => {
  afterEach(cleanup)

  it('renders', () => {
    const { container } = renderWithReactIntl(
      <Mobsters
        activeUsers={activeUsers}
        clickEditButton={jest.fn()}
        clickGitHubButton={jest.fn()}
        clickGuestButton={jest.fn()}
        clickRemoveUser={jest.fn()}
        inactiveUsers={inactiveUsers}
        onDragEnd={jest.fn()}
        setUsername={jest.fn()}
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders when the user is editing the list', () => {
    const { container } = renderWithReactIntl(
      <Mobsters
        activeUsers={activeUsers}
        clickEditButton={jest.fn()}
        clickGitHubButton={jest.fn()}
        clickGuestButton={jest.fn()}
        clickRemoveUser={jest.fn()}
        inactiveUsers={inactiveUsers}
        isEditing={true}
        onDragEnd={jest.fn()}
        setUsername={jest.fn()}
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders when there is a username', () => {
    const { container } = renderWithReactIntl(
      <Mobsters
        activeUsers={activeUsers}
        clickEditButton={jest.fn()}
        clickGitHubButton={jest.fn()}
        clickGuestButton={jest.fn()}
        clickRemoveUser={jest.fn()}
        inactiveUsers={inactiveUsers}
        isEditing={true}
        onDragEnd={jest.fn()}
        setUsername={jest.fn()}
        username="Some user"
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
