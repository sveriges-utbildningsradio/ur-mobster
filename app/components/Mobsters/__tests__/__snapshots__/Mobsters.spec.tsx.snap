// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Mobsters renders 1`] = `
<div
  class="wrap"
>
  <div>
    <div>
      <div
        class="headerWrap"
      >
        <h4>
          Mobsters
        </h4>
        <span
          class="editLabel"
        >
          <span>
            Redigera
          </span>
        </span>
        <input
          alt="Edit mobsters"
          class="editButton"
          data-e2e="mobsters-edit-button"
          data-testid="mobsters-edit-button"
          src="test-file-stub"
          type="image"
        />
      </div>
    </div>
    <ul>
      <div
        class="droppableWrap"
        data-react-beautiful-dnd-droppable="0"
        style="background-color: rgb(17, 6, 54);"
      >
        <ul
          data-e2e="mobsters-list-draggable-wrapper-activeUsers"
        >
          <li
            aria-roledescription="Draggable item. Press space bar to lift"
            class="mobster"
            data-react-beautiful-dnd-drag-handle="0"
            data-react-beautiful-dnd-draggable="0"
            draggable="false"
            tabindex="0"
          >
            <div
              class="avatarWrap"
            >
              <img
                alt="Active 1 Name"
                class="avatar"
                src="https://avatars1.githubusercontent.com/u/test"
              />
              <span
                class="driverDot"
              />
            </div>
            <div>
              <p
                class="name"
                data-e2e="mobsterslist-name-0"
                data-testid="mobsterslist-name-0"
              >
                Active 1 Name
              </p>
              <p
                class="githubName"
                data-e2e="mobsterslist-githubName-0"
                data-testid="mobsterslist-githubName-0"
              >
                Active 1 Githubname
              </p>
            </div>
          </li>
          <li
            aria-roledescription="Draggable item. Press space bar to lift"
            class="mobster"
            data-react-beautiful-dnd-drag-handle="0"
            data-react-beautiful-dnd-draggable="0"
            draggable="false"
            tabindex="0"
          >
            <div
              class="avatarWrap"
            >
              <img
                alt="Active 2 Name"
                class="avatar"
                src="test-file-stub"
              />
              <span
                class="navigatorDot"
              />
            </div>
            <div>
              <p
                class="name"
                data-e2e="mobsterslist-name-1"
                data-testid="mobsterslist-name-1"
              >
                Active 2 Name
              </p>
              <p
                class="githubName"
                data-e2e="mobsterslist-githubName-1"
                data-testid="mobsterslist-githubName-1"
              />
            </div>
          </li>
        </ul>
      </div>
    </ul>
    <h4>
      Inaktiva mobsters
    </h4>
    <ul>
      <div
        class="droppableWrap"
        data-react-beautiful-dnd-droppable="0"
        style="background-color: rgb(17, 6, 54);"
      >
        <ul
          data-e2e="mobsters-list-draggable-wrapper-inactiveUsers"
        >
          <li
            aria-roledescription="Draggable item. Press space bar to lift"
            class="mobster"
            data-react-beautiful-dnd-drag-handle="0"
            data-react-beautiful-dnd-draggable="0"
            draggable="false"
            tabindex="0"
          >
            <div
              class="avatarWrap"
            >
              <img
                alt="Inactive 2 Name"
                class="avatar"
                src="https://avatars1.githubusercontent.com/u/inactivetest"
              />
            </div>
            <div>
              <p
                class="name"
                data-e2e="mobsterslist-name-0"
                data-testid="mobsterslist-name-0"
              >
                Inactive 2 Name
              </p>
              <p
                class="githubName"
                data-e2e="mobsterslist-githubName-0"
                data-testid="mobsterslist-githubName-0"
              >
                Inactive 1 Githubname
              </p>
            </div>
          </li>
          <li
            aria-roledescription="Draggable item. Press space bar to lift"
            class="mobster"
            data-react-beautiful-dnd-drag-handle="0"
            data-react-beautiful-dnd-draggable="0"
            draggable="false"
            tabindex="0"
          >
            <div
              class="avatarWrap"
            >
              <img
                alt="Inactive 2 Name"
                class="avatar"
                src="test-file-stub"
              />
            </div>
            <div>
              <p
                class="name"
                data-e2e="mobsterslist-name-1"
                data-testid="mobsterslist-name-1"
              >
                Inactive 2 Name
              </p>
              <p
                class="githubName"
                data-e2e="mobsterslist-githubName-1"
                data-testid="mobsterslist-githubName-1"
              />
            </div>
          </li>
        </ul>
      </div>
    </ul>
  </div>
  <div
    class="addWrap"
  >
    <img
      alt="Default avatar"
      class="avatar"
      src="test-file-stub"
    />
    <div>
      <div
        class="inputWrap"
      >
        <input
          class="userInput"
          data-e2e="mobsters-add-input"
          placeholder="Lägg till mobster"
          value=""
        />
        <button
          class="addButton"
          data-e2e="mobsters-add-by-name"
          disabled=""
          type="button"
        >
          <img
            alt="Lägg till mobster som gäst"
            class="addButtonImage"
            src="test-file-stub"
          />
        </button>
        <button
          class="addButton"
          data-e2e="mobsters-add-by-github-username"
          disabled=""
          src="test-file-stub"
          type="button"
        >
          <img
            alt="Lägg till mobster med GitHub-konto"
            class="addButtonImage"
            src="test-file-stub"
          />
        </button>
      </div>
      <p
        class="githubName"
      >
        <span>
          Skriv namn eller GitHub-användarnamn
        </span>
      </p>
    </div>
  </div>
</div>
`;

exports[`components/Mobsters renders when the user is editing the list 1`] = `
<div
  class="wrap"
>
  <div>
    <div>
      <div
        class="headerWrap"
      >
        <h4>
          Mobsters
        </h4>
        <span
          class="editLabel"
        >
          <span>
            Spara
          </span>
        </span>
        <input
          alt="Stop editing mobsters"
          class="editButton"
          data-e2e="mobsters-edit-button"
          data-testid="mobsters-edit-button"
          src="test-file-stub"
          type="image"
        />
      </div>
    </div>
    <ul>
      <div
        class="droppableWrap"
        data-react-beautiful-dnd-droppable="1"
        style="background-color: rgb(17, 6, 54);"
      >
        <ul
          data-e2e="mobsters-list-draggable-wrapper-activeUsers"
        >
          <li
            aria-roledescription="Draggable item. Press space bar to lift"
            class="mobster"
            data-react-beautiful-dnd-drag-handle="1"
            data-react-beautiful-dnd-draggable="1"
            draggable="false"
            tabindex="0"
          >
            <div
              class="avatarWrap"
            >
              <img
                alt="Active 1 Name"
                class="avatar"
                src="https://avatars1.githubusercontent.com/u/test"
              />
              <span
                class="driverDot"
              />
            </div>
            <div>
              <p
                class="name"
                data-e2e="mobsterslist-name-0"
                data-testid="mobsterslist-name-0"
              >
                Active 1 Name
              </p>
              <p
                class="githubName"
                data-e2e="mobsterslist-githubName-0"
                data-testid="mobsterslist-githubName-0"
              >
                Active 1 Githubname
              </p>
            </div>
            <input
              alt="Remove user from activeUsers"
              class="editButton"
              data-e2e="mobsterslist-remove-user"
              data-testid="mobsterslist-remove-0"
              src="test-file-stub"
              type="image"
            />
          </li>
          <li
            aria-roledescription="Draggable item. Press space bar to lift"
            class="mobster"
            data-react-beautiful-dnd-drag-handle="1"
            data-react-beautiful-dnd-draggable="1"
            draggable="false"
            tabindex="0"
          >
            <div
              class="avatarWrap"
            >
              <img
                alt="Active 2 Name"
                class="avatar"
                src="test-file-stub"
              />
              <span
                class="navigatorDot"
              />
            </div>
            <div>
              <p
                class="name"
                data-e2e="mobsterslist-name-1"
                data-testid="mobsterslist-name-1"
              >
                Active 2 Name
              </p>
              <p
                class="githubName"
                data-e2e="mobsterslist-githubName-1"
                data-testid="mobsterslist-githubName-1"
              />
            </div>
            <input
              alt="Remove user from activeUsers"
              class="editButton"
              data-e2e="mobsterslist-remove-user"
              data-testid="mobsterslist-remove-1"
              src="test-file-stub"
              type="image"
            />
          </li>
        </ul>
      </div>
    </ul>
    <h4>
      Inaktiva mobsters
    </h4>
    <ul>
      <div
        class="droppableWrap"
        data-react-beautiful-dnd-droppable="1"
        style="background-color: rgb(17, 6, 54);"
      >
        <ul
          data-e2e="mobsters-list-draggable-wrapper-inactiveUsers"
        >
          <li
            aria-roledescription="Draggable item. Press space bar to lift"
            class="mobster"
            data-react-beautiful-dnd-drag-handle="1"
            data-react-beautiful-dnd-draggable="1"
            draggable="false"
            tabindex="0"
          >
            <div
              class="avatarWrap"
            >
              <img
                alt="Inactive 2 Name"
                class="avatar"
                src="https://avatars1.githubusercontent.com/u/inactivetest"
              />
            </div>
            <div>
              <p
                class="name"
                data-e2e="mobsterslist-name-0"
                data-testid="mobsterslist-name-0"
              >
                Inactive 2 Name
              </p>
              <p
                class="githubName"
                data-e2e="mobsterslist-githubName-0"
                data-testid="mobsterslist-githubName-0"
              >
                Inactive 1 Githubname
              </p>
            </div>
            <input
              alt="Remove user from inactiveUsers"
              class="editButton"
              data-e2e="mobsterslist-remove-user"
              data-testid="mobsterslist-remove-0"
              src="test-file-stub"
              type="image"
            />
          </li>
          <li
            aria-roledescription="Draggable item. Press space bar to lift"
            class="mobster"
            data-react-beautiful-dnd-drag-handle="1"
            data-react-beautiful-dnd-draggable="1"
            draggable="false"
            tabindex="0"
          >
            <div
              class="avatarWrap"
            >
              <img
                alt="Inactive 2 Name"
                class="avatar"
                src="test-file-stub"
              />
            </div>
            <div>
              <p
                class="name"
                data-e2e="mobsterslist-name-1"
                data-testid="mobsterslist-name-1"
              >
                Inactive 2 Name
              </p>
              <p
                class="githubName"
                data-e2e="mobsterslist-githubName-1"
                data-testid="mobsterslist-githubName-1"
              />
            </div>
            <input
              alt="Remove user from inactiveUsers"
              class="editButton"
              data-e2e="mobsterslist-remove-user"
              data-testid="mobsterslist-remove-1"
              src="test-file-stub"
              type="image"
            />
          </li>
        </ul>
      </div>
    </ul>
  </div>
  <div
    class="addWrap"
  >
    <img
      alt="Default avatar"
      class="avatar"
      src="test-file-stub"
    />
    <div>
      <div
        class="inputWrap"
      >
        <input
          class="userInput"
          data-e2e="mobsters-add-input"
          placeholder="Lägg till mobster"
          value=""
        />
        <button
          class="addButton"
          data-e2e="mobsters-add-by-name"
          disabled=""
          type="button"
        >
          <img
            alt="Lägg till mobster som gäst"
            class="addButtonImage"
            src="test-file-stub"
          />
        </button>
        <button
          class="addButton"
          data-e2e="mobsters-add-by-github-username"
          disabled=""
          src="test-file-stub"
          type="button"
        >
          <img
            alt="Lägg till mobster med GitHub-konto"
            class="addButtonImage"
            src="test-file-stub"
          />
        </button>
      </div>
      <p
        class="githubName"
      >
        <span>
          Skriv namn eller GitHub-användarnamn
        </span>
      </p>
    </div>
  </div>
</div>
`;

exports[`components/Mobsters renders when there is a username 1`] = `
<div
  class="wrap"
>
  <div>
    <div>
      <div
        class="headerWrap"
      >
        <h4>
          Mobsters
        </h4>
        <span
          class="editLabel"
        >
          <span>
            Spara
          </span>
        </span>
        <input
          alt="Stop editing mobsters"
          class="editButton"
          data-e2e="mobsters-edit-button"
          data-testid="mobsters-edit-button"
          src="test-file-stub"
          type="image"
        />
      </div>
    </div>
    <ul>
      <div
        class="droppableWrap"
        data-react-beautiful-dnd-droppable="2"
        style="background-color: rgb(17, 6, 54);"
      >
        <ul
          data-e2e="mobsters-list-draggable-wrapper-activeUsers"
        >
          <li
            aria-roledescription="Draggable item. Press space bar to lift"
            class="mobster"
            data-react-beautiful-dnd-drag-handle="2"
            data-react-beautiful-dnd-draggable="2"
            draggable="false"
            tabindex="0"
          >
            <div
              class="avatarWrap"
            >
              <img
                alt="Active 1 Name"
                class="avatar"
                src="https://avatars1.githubusercontent.com/u/test"
              />
              <span
                class="driverDot"
              />
            </div>
            <div>
              <p
                class="name"
                data-e2e="mobsterslist-name-0"
                data-testid="mobsterslist-name-0"
              >
                Active 1 Name
              </p>
              <p
                class="githubName"
                data-e2e="mobsterslist-githubName-0"
                data-testid="mobsterslist-githubName-0"
              >
                Active 1 Githubname
              </p>
            </div>
            <input
              alt="Remove user from activeUsers"
              class="editButton"
              data-e2e="mobsterslist-remove-user"
              data-testid="mobsterslist-remove-0"
              src="test-file-stub"
              type="image"
            />
          </li>
          <li
            aria-roledescription="Draggable item. Press space bar to lift"
            class="mobster"
            data-react-beautiful-dnd-drag-handle="2"
            data-react-beautiful-dnd-draggable="2"
            draggable="false"
            tabindex="0"
          >
            <div
              class="avatarWrap"
            >
              <img
                alt="Active 2 Name"
                class="avatar"
                src="test-file-stub"
              />
              <span
                class="navigatorDot"
              />
            </div>
            <div>
              <p
                class="name"
                data-e2e="mobsterslist-name-1"
                data-testid="mobsterslist-name-1"
              >
                Active 2 Name
              </p>
              <p
                class="githubName"
                data-e2e="mobsterslist-githubName-1"
                data-testid="mobsterslist-githubName-1"
              />
            </div>
            <input
              alt="Remove user from activeUsers"
              class="editButton"
              data-e2e="mobsterslist-remove-user"
              data-testid="mobsterslist-remove-1"
              src="test-file-stub"
              type="image"
            />
          </li>
        </ul>
      </div>
    </ul>
    <h4>
      Inaktiva mobsters
    </h4>
    <ul>
      <div
        class="droppableWrap"
        data-react-beautiful-dnd-droppable="2"
        style="background-color: rgb(17, 6, 54);"
      >
        <ul
          data-e2e="mobsters-list-draggable-wrapper-inactiveUsers"
        >
          <li
            aria-roledescription="Draggable item. Press space bar to lift"
            class="mobster"
            data-react-beautiful-dnd-drag-handle="2"
            data-react-beautiful-dnd-draggable="2"
            draggable="false"
            tabindex="0"
          >
            <div
              class="avatarWrap"
            >
              <img
                alt="Inactive 2 Name"
                class="avatar"
                src="https://avatars1.githubusercontent.com/u/inactivetest"
              />
            </div>
            <div>
              <p
                class="name"
                data-e2e="mobsterslist-name-0"
                data-testid="mobsterslist-name-0"
              >
                Inactive 2 Name
              </p>
              <p
                class="githubName"
                data-e2e="mobsterslist-githubName-0"
                data-testid="mobsterslist-githubName-0"
              >
                Inactive 1 Githubname
              </p>
            </div>
            <input
              alt="Remove user from inactiveUsers"
              class="editButton"
              data-e2e="mobsterslist-remove-user"
              data-testid="mobsterslist-remove-0"
              src="test-file-stub"
              type="image"
            />
          </li>
          <li
            aria-roledescription="Draggable item. Press space bar to lift"
            class="mobster"
            data-react-beautiful-dnd-drag-handle="2"
            data-react-beautiful-dnd-draggable="2"
            draggable="false"
            tabindex="0"
          >
            <div
              class="avatarWrap"
            >
              <img
                alt="Inactive 2 Name"
                class="avatar"
                src="test-file-stub"
              />
            </div>
            <div>
              <p
                class="name"
                data-e2e="mobsterslist-name-1"
                data-testid="mobsterslist-name-1"
              >
                Inactive 2 Name
              </p>
              <p
                class="githubName"
                data-e2e="mobsterslist-githubName-1"
                data-testid="mobsterslist-githubName-1"
              />
            </div>
            <input
              alt="Remove user from inactiveUsers"
              class="editButton"
              data-e2e="mobsterslist-remove-user"
              data-testid="mobsterslist-remove-1"
              src="test-file-stub"
              type="image"
            />
          </li>
        </ul>
      </div>
    </ul>
  </div>
  <div
    class="addWrap"
  >
    <img
      alt="Default avatar"
      class="avatar"
      src="test-file-stub"
    />
    <div>
      <div
        class="inputWrap"
      >
        <input
          class="userInput"
          data-e2e="mobsters-add-input"
          placeholder="Lägg till mobster"
          value="Some user"
        />
        <button
          class="addButton"
          data-e2e="mobsters-add-by-name"
          type="button"
        >
          <img
            alt="Lägg till mobster som gäst"
            class="addButtonImage"
            src="test-file-stub"
          />
        </button>
        <button
          class="addButton"
          data-e2e="mobsters-add-by-github-username"
          src="test-file-stub"
          type="button"
        >
          <img
            alt="Lägg till mobster med GitHub-konto"
            class="addButtonImage"
            src="test-file-stub"
          />
        </button>
      </div>
      <p
        class="githubName"
      >
        <span>
          Skriv namn eller GitHub-användarnamn
        </span>
      </p>
    </div>
  </div>
</div>
`;
